Sources/Configuration/DefaultConfigurationManager.swift

```diff
@@ -0,0 +1,160 @@
+//
+
+import Foundation
+import os.log
+import Combine
+import Common
+import Persistence
+
+public extension Logger {
+    static var config: Logger = { Logger(subsystem: Bundle.main.bundleIdentifier ?? "DuckDuckGo", category: "Configuration") }()
+}
+
+open class DefaultConfigurationManager: NSObject {
+    public enum Error: Swift.Error {
+
+        case timeout
+        case bloomFilterSpecNotFound
+        case bloomFilterBinaryNotFound
+        case bloomFilterPersistenceFailed
+        case bloomFilterExclusionsNotFound
+        case bloomFilterExclusionsPersistenceFailed
+
+        public func withUnderlyingError(_ underlyingError: Swift.Error) -> Swift.Error {
+            let nsError = self as NSError
+            return NSError(domain: nsError.domain, code: nsError.code, userInfo: [NSUnderlyingErrorKey: underlyingError])
+        }
+
+    }
+
+    public enum Constants {
+
+        public static let downloadTimeoutSeconds = 60.0 * 5
+#if DEBUG
+        public static let refreshPeriodSeconds = 60.0 * 2 // 2 minutes
+#else
+        public static let refreshPeriodSeconds = 60.0 * 30 // 30 minutes
+#endif
+        public static let retryDelaySeconds = 60.0 * 60 * 1 // 1 hour delay before checking again if something went wrong last time
+        public static let refreshCheckIntervalSeconds = 60.0 // check if we need a refresh every minute
+
+        static let lastUpdateDefaultsKey = "configuration.lastUpdateTime"
+    }
+
+    private var defaults: KeyValueStoring
+
+    public var fetcher: ConfigurationFetching
+    public var store: ConfigurationStoring
+
+    public init(fetcher: ConfigurationFetching, store: ConfigurationStoring, defaults: KeyValueStoring = UserDefaults()) {
+        self.fetcher = fetcher
+        self.store = store
+        self.defaults = defaults
+        super.init()
+        NSFileCoordinator.addFilePresenter(self)
+    }
+
+    deinit {
+        NSFileCoordinator.removeFilePresenter(self)
+    }
+
+    public static let queue: DispatchQueue = DispatchQueue(label: "Configuration Manager")
+    public static let filePresenterOperationQueue = OperationQueue()
+
+    public var lastUpdateTime: Date {
+        get {
+            defaults.object(forKey: Constants.lastUpdateDefaultsKey) as? Date ?? .distantPast
+        }
+        set {
+            defaults.set(newValue, forKey: Constants.lastUpdateDefaultsKey)
+        }
+    }
+
+    private var timerCancellable: AnyCancellable?
+    public var lastRefreshCheckTime: Date = Date()
+
+    public func start() {
+        Logger.config.debug("Starting configuration refresh timer")
+        timerCancellable = Timer.publish(every: Constants.refreshCheckIntervalSeconds, on: .main, in: .default)
+            .autoconnect()
+            .receive(on: Self.queue)
+            .sink(receiveValue: { _ in
+                self.lastRefreshCheckTime = Date()
+                self.refreshIfNeeded()
+            })
+        Task {
+            await refreshNow()
+        }
```

Comment:
`NSTimer` won't keep time consistently in a network extension environment (Apple forum thread about it [here](https://forums.developer.apple.com/forums/thread/687170)), but given that we don't need perfect timing I'd say it's acceptable. What do you think @diegoreymendez?